ocaml/xapi/workload_balancing.ml

```diff
@@ -114,6 +115,66 @@ let assert_wlb_enabled ~__context =
   if not (Db.Pool.get_wlb_enabled ~__context ~self:pool) then
     raise_disabled ()
 
+let read_file path =
+  let ic = open_in path in
+  Fun.protect
+    ~finally:(fun () -> close_in ic ; Sys.remove path)
+    (fun () ->
+      let length = in_channel_length ic in
+      really_input_string ic length
+    )
+
+let fetch_certificate wlb_url =
+  Ssl.init () ;
+
+  let ( >>= ) = Result.bind in
+
+  let parse_url wlb_url =
+    match String.split_on_char ':' wlb_url with
+    | [host; port] ->
+        Ok (host, int_of_string port)
+    | _ ->
+        Error "Invalid URL format"
+  in
+
+  let open_connection (host, port) =
+    let ctx = Ssl.create_context Ssl.TLSv1_2 Ssl.Client_context in
+    try
+      Ok
+        (Ssl.open_connection_with_context ctx
+           (Unix.ADDR_INET (Unix.inet_addr_of_string host, port))
+        )
+    with _ -> Error "Failed to connect to the server"
+  in
+
+  let fetch_and_validate_cert ssl_socket =
```

Comment:
Could this be done in a separate external process? Then we don't need to link `ssl` with XAPI.
This doesn't appear to be a performance critical section, so spawning another process via forkhelpers shouldn't be a problem.

The separate process could still be built as part of the XAPI repo.

Reply:
One possible solution that avoid link OpenSSL to XAPI may be getting the cert by OpenSSL toolkit, e.g. `openssl s_client -connect IP:PORT -showcerts`? then it will be done in a separate external process and not link ssl to XAPI as well.